HZETRN/btrn.py

```diff
@@ -0,0 +1,377 @@
+import hzetrn as one_d_tool
+import tag_utils
+import logging
+
+
+###############################
+# globals
+config_log  = 'config.log'
+# 'spatial_dir' is a subdir for storing input files as they are created
+spatial_dir = 'spatial'
+
+"""
+    Read the coordinates of the Reference Point from a file.
+    filename = name of file containing triplet
+        If filename = '' the reference point is set
+        to the origin
+"""
+def load_ray_start(filename):
+    # Set the Reference Point.
+    ray_start = []
+    if not filename:
+        return [0.0, 0.0, 0.0]
+    else:
+        with open(filename) as f:
+	    return map(float, f.readline().split())
+
+"""
+Load or create tuples representing dirs
+In file with 10,000 tuples, here are the ranges
+('min_theta', 0.00117047, 'max_theta', 3.136877355, 'min_phi', -3.141219364, 'max_phi', 3.14087921)
+
+filename name of file with unit direction vectors 
+         Its existence is checked before entering this function.
+         The first member of each pair is the polar angle theta.
+	 The second member is the azimuthal angle phi.
+"""
+def load_ray_tuples(filename):
+
+    ray_tuples = []
+    
+    with open(filename) as f:
+        for line in f:
+	    nums = map(float, line.split())
+	    z = np.cos(nums[0])
+	    x = np.sin(nums[0])*np.cos(nums[1])
+	    y = np.sin(nums[0])*np.sin(nums[1])
+	    ray_tuples.append((x, y, z))
+    return ray_tuples
+
+"""
+Return an array of directions
+args =  parsed command line parameters
+        args.ray_dir_file, if not '', contains the directions
+	Otherwis, use random directons or Cartesian axes
+
+The set of directions is returned as a numpy row array
+"""
+def get_directions(ray_dir_file, num_rand_dirs):
+    # Return the ray directions 
+    if ray_dir_file:
+        ray_tuples = load_ray_tuples(ray_dir_file)
+
+    elif num_rand_dirs > 0:
+        ray_tuples = tag_utils.get_rand_dirs(num_rand_dirs)
+ 
+    else:
+        ray_tuples = [(1.0, 0.0, 0.0),
+                      (0.0, 1.0, 0.0),
+	   	      (0.0, 0.0, 1.0)]
+    return np.array(ray_tuples)
+
+
+""" 
+Find the vol-id of the geometry that contains the ref_point
+Requires dagmc python library to be loaded
+"""
+def find_ref_vol(ref_point):
+    # code snippet from dagmc.pyx:find_graveyard_inner_box
+    volumes = dagmc.get_volume_list() 
+    for v in volumes:
+        if dagmc.point_in_volume(v, ref_point):
+            return v
+
+"""
+Find the slab distances and material names
+start_vol  The volume containing the ref_point
+ref_point  the origin of the ray
+dir 	   the direction of the ray
+fname_for_vol a mapping of vol id to material object
+
+returns    arrays  for length and material name as the ray
+           travels from the start point to the graveyard
+
+This depends on dagmc's ray_iterator, which calls ray_fire
+"""
+def slabs_for_ray(start_vol, ref_point, dir, fname_for_vol):
+    is_graveyard = False
+    foundMat = False
+    surf = 0
+    dist = 0
+    huge = 1000000000
```

Comment:
can we do like 1.0e308 or something? 1.0e299 - https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0CB4QFjAA&url=http%3A%2F%2Fstackoverflow.com%2Fquestions%2F3477283%2Fmaximum-float-in-python&ei=Y81LVc2kGdXXoASfjoHICQ&usg=AFQjCNHspk9ywcYFaF8lv9N4BUN3tHIUJw&sig2=8kt0Gvt9Z0uY_GuJYTm7MA&bvm=bv.92765956,d.aWw